### PR TITLE
net_plugin fix for privacy groups corner case

### PR DESCRIPTION
**Pull request by dimas1185**
## Change Description

Fixed test case that required node restart in `privacy_scenario_3_test.py`

Scenario:
after `node1` launched it sends handshake to `node2`
`node1` is not in security group so `node2` doesn't send back any notice (lib catchup notice was required in that case)
`node1` was added to security group and `node2` sends to `node1` handshake
in `sync 1` scenario, when `net_plugin` receives non-first handshake with lib greater its own lib it do nothing. 
This fix adds sending handshake there also in case of non-first handshake generation

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [x] Stability bug fix

- [ ] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [x] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Review by brianjohnson5972 — CHANGES_REQUESTED** [review]
I'd also like Kevin to look at this.

File: plugins/net_plugin/net_plugin.cpp [hunk 1]
```diff
@@ -2056,6 +2056,9 @@ namespace eosio {
                   ("ep", c->peer_name())("lib", msg.last_irreversible_block_num)("head", msg.head_num)
                   ("id", msg.head_id.str().substr(8,16)) );
          c->syncing = false;
+         if (c->sent_handshake_count > 0) {
```
[the remark this answers is not in the document]
Comment (dimas1185) [reply plugins/net_plugin/net_plugin.cpp:2059]: because of this we will send it twice then:
```
void connection::handle_message( const handshake_message& msg ) {
...
      if (msg.generation == 1) {
      ...
         if( sent_handshake_count == 0 ) {
            send_handshake();
         }
      }
      ...
      my_impl->sync_master->recv_handshake( shared_from_this(), msg );

}
```
As an alternative, since we process `recv_handshake` anyway, we can remove it from `handle_message` and make unconditional inside `recv_handshake`.
What do you think?

**Review by heifner — APPROVED** [review]

**Review by brianjohnson5972 — APPROVED** [review]
